src/methods/wallet-get-calls-status.ts

```diff
@@ -1,85 +1,69 @@
 import { rpcErrors } from '@metamask/rpc-errors';
 import type { Infer } from '@metamask/superstruct';
-import {
-  nonempty,
-  optional,
-  mask,
-  string,
-  array,
-  object,
-  tuple,
-} from '@metamask/superstruct';
+import { tuple } from '@metamask/superstruct';
 import type {
+  Hex,
   Json,
   JsonRpcRequest,
   PendingJsonRpcResponse,
 } from '@metamask/utils';
-import { HexChecksumAddressStruct, StrictHexStruct } from '@metamask/utils';
+import { StrictHexStruct } from '@metamask/utils';
 
 import { validateParams } from '../utils/validation';
 
-const GetCallsStatusStruct = tuple([nonempty(string())]);
+const GetCallsStatusStruct = tuple([StrictHexStruct]);
 
-const GetCallsStatusReceiptStruct = object({
-  logs: optional(
-    array(
-      object({
-        address: optional(HexChecksumAddressStruct),
-        data: optional(StrictHexStruct),
-        topics: optional(array(StrictHexStruct)),
-      }),
-    ),
-  ),
-  status: optional(StrictHexStruct),
-  chainId: optional(StrictHexStruct),
-  blockHash: optional(StrictHexStruct),
-  blockNumber: optional(StrictHexStruct),
-  gasUsed: optional(StrictHexStruct),
-  transactionHash: optional(StrictHexStruct),
-});
+export enum GetCallsStatusCode {
+  PENDING = 100,
+  CONFIRMED = 200,
+  FAILED_OFFCHAIN = 400,
+  REVERTED = 500,
+  REVERTED_PARTIAL = 600,
+}
 
 export type GetCallsStatusParams = Infer<typeof GetCallsStatusStruct>;
-export type GetCallsStatusReceipt = Infer<typeof GetCallsStatusReceiptStruct>;
 
 export type GetCallsStatusResult = {
-  status: 'PENDING' | 'CONFIRMED';
-  receipts?: GetCallsStatusReceipt[];
+  version: string;
+  id: Hex;
+  chainId: Hex;
+  status: number;
```

Comment:
Should this be:

``` suggestion
  status: (typeof GetCallsStatusCode)[keyof typeof GetCallsStatusCode];
```

Reply:
The enum is a convenience for the common / current codes, but this was intentionally not coupled to it so the client is still free to return newer or even custom codes in future since the codes are grouped similar to HTTP response.

Reply:
Gotcha, makes sense.